Neos.ContentRepository.Core/Classes/Factory/ContentRepositoryFactory.php

```diff
@@ -117,11 +117,12 @@ public function build(): ContentRepository
      * @return T
      */
     public function buildService(
+        ContentRepository $contentRepository,
```

Comment:
I stumble upon this new parameter.
I know why it is required, but the `ContentRepositoryFactory` is already bound to a `ContentRepositoryId` and it seems weird to me that you could (in theory) pass a _different_ CR instance to that method.

I wonder if it might be more feasible to extract the service creation into a separate `ContentRepositoryServiceFactory` (and whether that would help at all)

Reply:
we could alternatively go through all `$this->factoryInstances` in `getContentRepositoryIdByContentRepository` and ask each factory if it has been build and if the cr belongs to this factory like:

```php
public function didBuild(ContentRepository $cr): bool
{
   return $this->contentRepository === $cr;
}
```

then knowing that the factory "owns", "mangages", "isFactoryForCr" or "didBuild" this cr, we know via the array key of `$this->factoryInstances` the cr id

Reply:
The thing is that a factory should just do this one thing, build instances.
IMO it should not magically turn objects into singletons or keep track of the built instances